engine/src/test/scala/cromwell/engine/workflow/lifecycle/execution/ejea/EjeaInvalidatingCacheEntrySpec.scala

```diff
@@ -17,49 +17,36 @@ class EjeaInvalidatingCacheEntrySpec extends EngineJobExecutionActorSpec with Ca
     val invalidateSuccess = CallCacheInvalidatedSuccess
     val invalidateFailure = CallCacheInvalidatedFailure(invalidationErrorCause)
 
-    val metaInfo31: MetaInfoId = MetaInfoId(31)
-    val metaInfo32: MetaInfoId = MetaInfoId(32)
+    val metaInfo31: CallCachingEntryId = CallCachingEntryId(31)
+    val metaInfo32: CallCachingEntryId = CallCachingEntryId(32)
     val cacheHitWithTwoIds = CacheHit(NonEmptyList.of(metaInfo32, metaInfo31))
     val cacheHitWithSingleId = CacheHit(NonEmptyList.of(metaInfo31))
-    val noCacheHit = None
-
-    case class CacheHitAndNextExpected(name: String,
-                                       cacheHitData: CacheHit,
-                                       nextCacheHitToTry: MetaInfoId,
-                                       nextExpectedCacheHitData: Option[CacheHit])
-
-    val cacheHitCombinations = List(
-      CacheHitAndNextExpected("cache hit with single Id hit", cacheHitWithSingleId, metaInfo31, None),
-      CacheHitAndNextExpected("cache hit with two Id hits", cacheHitWithTwoIds, metaInfo32, Option(cacheHitWithSingleId))
-    )
 
     RestartOrExecuteCommandTuples foreach { case RestartOrExecuteCommandTuple(operationName, restarting, expectedMessage) =>
       List(invalidateSuccess, invalidateFailure) foreach { invalidateActorResponse =>
         s"$operationName a job if cache invalidation succeeds and there are no other cache hits to try when invalidate response is $invalidateActorResponse" in {
-          ejea = ejeaInvalidatingCacheEntryState(noCacheHit, restarting = restarting)
+          ejea = ejeaInvalidatingCacheEntryState(Option(cacheHitWithSingleId), restarting = restarting)
           // Send the response from the invalidate actor
           ejea ! invalidateActorResponse
 
           helper.bjeaProbe.expectMsg(awaitTimeout, expectedMessage)
-          ejea.stateName should be(RunningJob)
+          eventually { ejea.stateName should be(RunningJob) }
           ejea.stateData should be(ResponsePendingData(helper.backendJobDescriptor, helper. bjeaProps, None, None))
         }
       }
     }
 
-    cacheHitCombinations foreach { case combo @ CacheHitAndNextExpected(hitComboName, initialHitData, nextHitToTry, finalHitData) =>
-        List(invalidateSuccess, invalidateFailure) foreach { invalidateActorResponse =>
-          s"try the next available hit when there is a $hitComboName when response is $invalidateActorResponse" in {
-            ejea = ejeaInvalidatingCacheEntryState(Option(initialHitData), restarting = false)
-            // Send the response from the invalidate actor
-            ejea ! invalidateActorResponse
+    List(invalidateSuccess, invalidateFailure) foreach { invalidateActorResponse =>
```

Comment:
If you invert the order of the two `foreach` blocks above, you could avoid having to have this second `foreach` here